pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -472,17 +480,26 @@ def save_checkpoint(self, checkpoint: Dict, filepath: str) -> None:
         """Save model/training states as a checkpoint file through state-dump and file-write.
 
         Args:
+            checkpoint: The checkpoint state dictionary
             filepath: write-target file's path
-            weights_only: saving model weights only
         """
         if self.world_size > 1 and self.zero_stage_3:
+            save_dir = self._filepath_to_dir(filepath)
+            if self.save_full_weights:
+                # todo: expose this as a unprotected function in deepspeed
+                state_dict = self.deepspeed_engine._zero3_consolidated_fp16_state_dict()
+                if self.is_global_zero:
+                    # Remove module portion of the checkpoint before saving
+                    state_dict = {k.partition('module.')[2]: state_dict[k] for k in state_dict.keys()}
+                    checkpoint['state_dict'] = state_dict
```

Comment:
n00b question: is this flipped? what does the state dict look like for non rank-0 ranks?

Reply:
What do you mean flipped? It's `None` on every other rank: https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/runtime/engine.py#L1713